sound/soc/sof/trace.c

```diff
@@ -566,12 +573,20 @@ void snd_sof_release_trace(struct snd_sof_dev *sdev)
 			dev_err(sdev->dev, "DMA_TRACE_FREE failed with error: %d\n", ret);
 	}
 
+	only_stop = (snd_sof_dsp_power_target(sdev) == SOF_DSP_PM_D0);
```

Comment:
the boolean variable only_stop is not needed, just if (snd_sof_dsp_power_target(sdev) == SOF_DSP_PM_D0) is good enough

Reply:
besides you have already stopped, so sdev->dtrace_state = SOF_DTRACE_STOPPED; needs to go before the if()

Reply:
and the target state should be _passed_ to the release function and not queried via the snd_sof_release_trace()